Workflow/Sources/AnyWorkflowConvertible.swift

```diff
@@ -147,3 +147,18 @@ struct OutputBlockingWorkflow<Child: AnyWorkflowConvertible>: Workflow {
             .rendered(in: context)
     }
 }
+
+// MARK: -
+
+extension AnyWorkflowConvertible where Output == Never {
+    /// A convenience for workflows that don't produce output but want to change the declared output type.
+    ///
+    /// This equivalent to `.mapOutput { switch $0 {} }` but avoids the compiler warning for unreachable code in the provided transform.
+    public func mapOutput<NewOutput>() -> AnyWorkflow<Rendering, NewOutput> {
```

Comment:
nit: mostly thinking 'out loud', but if for some reason type inference isn't 'working', i wonder if we might want to use a defaulted parameter here so it can be explicitly controlled? i think i convinced myself it's simpler as-is and we could always add that later, so no need to change anything.